pkg/storagev2/chunkstore.go

```diff
@@ -17,21 +17,42 @@ type Getter interface {
 	// Get a chunk by its swarm.Address. Returns the chunk associated with
 	// the address alongside with its postage stamp, or a storage.ErrNotFound
 	// if the chunk is not found.
+	// If the chunk has multiple stamps, then the first stamp is returned in this
+	// query. In order to deterministically get the stamp, use the GetterWithStamp
+	// interface.
 	Get(context.Context, swarm.Address) (swarm.Chunk, error)
 }
 
+// GetterWithStamp is the interface that provides a way to specify a postage
+// stamp to filter the Get query. This is required as we will support multiple stamps
+// on a chunk. As a result some components would need to query chunk and provide
+// the stamp to return by using the argument.
+type GetterWithStamp interface {
+	GetWithStamp(context.Context, swarm.Address, []byte) (swarm.Chunk, error)
+}
+
 // Putter is the interface that wraps the basic Put method.
 type Putter interface {
 	// Put a chunk into the store alongside with its postage stamp.
 	Put(context.Context, swarm.Chunk) error
 }
 
-// Deleter is the interface that wraps the basic Delete method.
+// Deleter is the interface that wraps the basic Delete method. This delete will
+// remove all the stamps associated with the chunk. To delete based on stamps,
+// use the DeleterWithStamp interface.
 type Deleter interface {
 	// Delete a chunk by the given swarm.Address.
 	Delete(context.Context, swarm.Address) error
 }
 
+// DeleterWithStamp provides a way to Delete a chunk by specifying the stamp
```

Comment:
@istae @mrekucci 

So I thought about this, but it will be difficult to do the DeleteWithStamp function. The problem is we have a ref counted chunkstore. This means a chunk can be put with the same stamp by more than 1 component. So deleting the stamp when requested by 1 component is not a good idea.

Instead, the ref counting will handle the deletion properly. RefCnt is added everytime a Put is done. So a delete will only delete the Chunk if refCnt goes to 0. This means all the stamps will be kept in chunkstore till the chunk reaches refCnt 0.

We only need the GetterWithStamp interface. Only the upload store and reserve store will need to use this when we are doing syncing. So these components need to track the BatchIDs they are interested in and then use GetterWithStamp to get the right stamp before interacting with the syncing protocols. So I am removing this interface for now. If we need this functionality in future we can think about doing this.